src/libcadet/model/parts/TwoDimensionalConvectionDispersionOperator.cpp

```diff
@@ -729,6 +729,7 @@ bool TwoDimensionalConvectionDispersionOperator::configureModelDiscretization(IP
 //	_radialCentroids.resize(nRad);
 	_crossSections.resize(nRad);
 	_curVelocity.resize(nRad);
+	_dir.resize(nRad);
```

Comment:
This is not necessary as you overwrite the full vector later on. Doing
```cpp
_dir = std::vector<int>(_nRad, 1);
```
means to 
1. throw away the current vector `_dir` (deallocate memory if it has capacity `> 0`),
2. construct a new `std::vector<int>` that contains `_nRad` elements all with value `1`, and
3. assign the new vector to `_dir`. 

When you do `_dir.resize(nRad)`, you're telling the vector that you want to reserve memory for `nRad` elements. The vector is still empty, but it doesn't need to allocate memory if you put in (up to) `nRad` elements. So you're allocating memory here and throw it away later when you do `_dir = std::vector<int>(...)`. We can avoid this by not `resize()`ing the `vector`.

Writing `_dir(0)` in the constructor tells the `vector` that you want it to hold `0` items initially. Hopefully, it does not allocate memory and sets its capacity to `0` also.

We call `resize()` on the other vectors (e.g., `_curVelocity`) since we are not replacing them with new vectors, but using indexed access. That is, we need the vectors to be non-empty.